File: Data Structures & Algorithms/target-sum/submission-1.py
class Solution:
    def findTargetSumWays(self, nums: List[int], target: int) -> int:
        n = len(nums)

        def dfs(i,result):
            if i==n:
                return 1 if result==0 else 0
            
            ways = dfs(i+1, result+nums[i])+dfs(i+1,result-nums[i])

            return ways
        
        return dfs(0,target)

File: Data Structures & Algorithms/two-integer-sum/submission-1.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        for(int i = 0; i<nums.size(); i++){
            int complement = target-nums[i];
            auto it = std::find(nums.begin()+i+1, nums.end(), complement);
            if (it!= nums.end()){
                return {i, static_cast<int>(std::distance(nums.begin(), it))};
            }
        }
        return {};
    }
};
